src/osd/PrimaryLogPG.cc

```diff
@@ -1898,20 +1898,19 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
   }
 
   // object name too long?
-  if (m->get_oid().name.size() > cct->_conf->osd_max_object_name_len) {
-    dout(4) << "do_op name is longer than "
-	    << cct->_conf->osd_max_object_name_len
-	    << " bytes" << dendl;
-    osd->reply_op_error(op, -ENAMETOOLONG);
-    return;
-  }
-  if (m->get_hobj().get_key().size() > cct->_conf->osd_max_object_name_len) {
-    dout(4) << "do_op locator is longer than "
-	    << cct->_conf->osd_max_object_name_len
-	    << " bytes" << dendl;
+  auto osd_max_object_name_len =
+    cct->_conf->get_val<uint64_t>("osd_max_object_name_len");
```

Comment:
Nope.. this is in the fast past (do_op), and get_val<> is slow.

let's put a hold on these conversions for now until we have a lighter-weight replacement that isn't slow.  The legacy_config_opts.h thing works pretty well and we dont' have a good concise replacement yet.